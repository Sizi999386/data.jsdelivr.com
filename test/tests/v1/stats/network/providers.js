const { makeEndpointSnapshotTests, setupSnapshots } = require('../../../../utils');

const periodOptions = [ 'day', 'week', 'month', 'year', 'all', undefined ];

describe('/v1/stats/network/providers', () => {
	before(() => {
		setupSnapshots(__filename);
	});

	makeStatsNetworkProvidersTests();
});

function makeStatsNetworkProvidersTests () {
	makeEndpointSnapshotTests('/v1/stats/network/providers{?continent,country,period}', {
		period: 'month',
	}, [
		{
			period: periodOptions,
		},
		{
			continent: 'EU',
			period: periodOptions,
		},
		{
			country: 'PL',
			period: periodOptions,
		},
	]);

	makeEndpointSnapshotTests('/v1/stats/network/providers{?continent,country,period}', {
		period: 'month',
	}, [
		{ continent: 'X' },
		{ country: 'X' },
		{ continent: 'EU', country: 'PL' },
		{ period: 'X' },
	], { status: 400 });
}
