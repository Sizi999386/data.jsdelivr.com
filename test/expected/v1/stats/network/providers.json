{
	"/v1/stats/network/providers?continent=EU&country=PL&period=month": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?continent=EU&period=all": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?continent=EU&period=day": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?continent=EU&period=month": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?continent=EU&period=week": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?continent=EU&period=year": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?continent=X&period=month": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?country=PL&period=all": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?country=PL&period=day": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?country=PL&period=month": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?country=PL&period=week": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?country=PL&period=year": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?country=X&period=month": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?period=X": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?period=all": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?period=day": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?period=month": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?period=week": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	},
	"/v1/stats/network/providers?period=year": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Bad Request. Visit https://github.com/jsdelivr/data.jsdelivr.com for documentation."
	}
}
